frontend/src/components/Moment.js

```diff
@@ -35,12 +98,51 @@ class Moment extends Component {
                     onClose={() => this.setState({ showDeleteModal: false })}/>
             {this.state.toHome && <Redirect to={'/'} />}
             { (props.currentUser ? props.currentUser.username === props.user.username : false) && <h1 className="delete-button" onClick={this.deleteMoment}>X</h1> }
+            {
+                (props.currentUser ? props.currentUser.username === props.user.username : false) &&
+                !editing &&
+                props.editable &&
+                <img
+                    className="edit-button"
+                    src={`http://${process.env.REACT_APP_IP}/res/edit.png`}
+                    alt="edit"
+                    onClick={this.onEditClick}
+                />
+            }
                 <ConditionalWrap
                     condition={this.props.momentId}
                     wrap={children =>  <Link to={`/moment/${props.momentId}`}>{children}</Link>}
                 >
                     <img className="Moment-thumbnail" src={props.image} alt="moment"/>
-                    <h1 className="top-caption">{ props.description }</h1>
+                    {
+                        editing ?
+                        <div className="moment-editor-container">
+                            <textarea
+                                ref={(textarea) => this.Textarea = textarea}
+                                className="moment-editor-input"
+                                type="text"
+                                value={newDesc}
+                                onChange={this.onEdit}/>
+                            {
+                                editError && <h1 className="edit-error">{editError}</h1>
+                            }
+                            <button
+                                className="moment-editor-btn"
+                                style={{ marginRight: '0.5em'}}
+                                onClick={this.onSave}
+                                disabled={((token) && (newDesc.length === 0 || props.description === newDesc)) || isEditing}>Save</button>
```

Comment:
same here for the boolean thing. No need to change it though, just a pointer. It really doesn't matter in what were doing